Duplicati.Library.Backend.Storx/StorxBackend.cs

```diff
@@ -0,0 +1,491 @@
+                    HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+                }
+                catch (System.Net.WebException wex)
+                {
+                    if (wex.Response is HttpWebResponse response && response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                        throw new FileMissingException(wex);
+                    else
+                        throw new Exception(getResponseBodyOnError(endpoint, wex));
+                }
+            }
+
+        }
+
+        public IList<ICommandLineArgument> SupportedCommands
+        {
+            get
+            {
+                return new List<ICommandLineArgument>(new ICommandLineArgument[] {
+                    new CommandLineArgument("auth-password", CommandLineArgument.ArgumentType.Password, Strings.StorxBackend.DescriptionAuthPasswordShort, Strings.StorxBackend.DescriptionAuthPasswordLong),
+                    new CommandLineArgument("auth-username", CommandLineArgument.ArgumentType.String, Strings.StorxBackend.DescriptionAuthUsernameShort, Strings.StorxBackend.DescriptionAuthUsernameLong)
+                });
+            }
+        }
+
+        public string Description
+        {
+            get { return Strings.StorxBackend.Description; }
+        }
+
+        public void Test()
+        {
+            this.TestList();
+        }
+
+        public void CreateFolder()
+        {
+           // GetFolder(1);
+        }
+
+        #endregion
+
+        #region IDisposable Members
+
+        public void Dispose()
+        {
+        }
+
+        #endregion
+
+
+        public string[] DNSName
+        {
+            get { return new string[] { new System.Uri(root).Host }; }
+        }
+
+        public void Get(string remotename, System.IO.Stream stream)
+        {
+            StorxFileList fl1 = GetFiles();
+            if (fl1.Files != null)
+            {
+                var fileID = fl1.Files.FindAll(x => (x.name + "." + x.type) == remotename)[0];
+                string endpoint = "storage/file/" + fileID.fileId;
+                var req = CreateRequest(endpoint);
+                req.Method = System.Net.WebRequestMethods.Http.Get;
+                var rreq = new Utility.AsyncHttpRequest(req);
+                using (var s = rreq.GetResponseStream())
+                    Library.Utility.Utility.CopyStream(s, stream);
+            }
+
+        }
+
+        public Task PutAsync(string remotename, System.IO.Stream stream, CancellationToken cancelToken)
+        {
+            string endpoint = root+"storage/folder/" + m_root_folder + "/upload";
+            try
+            {
+                byte[] formData = ReadToEnd(stream);
+                var webClient = new WebClient();
+                string boundary = "------------------------" + DateTime.Now.Ticks.ToString("x");
+                string contentType = "multipart/form-data; boundary=" + boundary;
```

Comment:
There should be some multipart support functions in the utility library, so we can avoid the hand-coded multipart form here.